host/sgx/switchless.c

```diff
@@ -100,8 +100,12 @@ oe_result_t oe_start_switchless_manager(
     // because the maximum parallelism is dictated by the latter for
     // synchronous ocalls. We may need to revisit this for asynchronous
     // calls later.
-    if (num_host_workers > enclave->num_bindings)
-        num_host_workers = (uint32_t)enclave->num_bindings;
+    oe_mutex_lock(&enclave->lock);
+    {
+        if (num_host_workers > enclave->num_bindings)
```

Comment:
It is not clear why this code needs to be guarded. num_bindings is never modified after the enclave has been constructed.

Reply:
Thank you. Good observation. Fixed.